src/source/source_cache.js

```diff
@@ -44,6 +45,9 @@ class SourceCache extends Evented {
     _maxTileCacheSize: ?number;
     _paused: boolean;
     _shouldReloadOnResume: boolean;
+    _needsFullPlacement: boolean;
+    _placementIDs: Array<number>;
+    _placementIndex: number;
```

Comment:
I'm wondering if it would be helpful to separate the placement state (and maybe even the placement methods) into a separate class or object. If it is, it could also be done at the style and tile level.

Reply:
That sounds like a good idea, but I don't have a clear idea of where it would be best to draw the boundaries. Could you push an implementation?